nbclient/util.py

```diff
@@ -4,9 +4,52 @@
 # Distributed under the terms of the Modified BSD License.
 
 import asyncio
+import sys
 import inspect
 
 
+def check_ipython():
+    # original from vaex/asyncio.py
+    IPython = sys.modules.get('IPython')
+    if IPython:
+        IPython_version = tuple(map(int, IPython.__version__.split('.')))
+        if IPython_version < (7, 0, 0):
+            raise RuntimeError(f'You are using IPython {IPython.__version__} while we require'
+                               '7.0.0+, please update IPython')
+
+
+def check_patch_tornado():
+    """If tornado is imported, add the patched asyncio.Future to its tuple of acceptable Futures"""
+    # original from vaex/asyncio.py
+    if 'tornado' in sys.modules:
+        import tornado.concurrent
+        if asyncio.Future not in tornado.concurrent.FUTURES:
+            tornado.concurrent.FUTURES = tornado.concurrent.FUTURES + (asyncio.Future, )
+
+
+def just_run(coro):
+    """Make the coroutine run, even if there is an event loop running (using nest_asyncio)"""
+    # original from vaex/asyncio.py
+    try:
+        loop = asyncio.get_event_loop()
+        had_loop = True
+    except RuntimeError:
+        loop = None
+        had_loop = False
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+    try:
+        if had_loop:
+            check_ipython()
+            import nest_asyncio
+            nest_asyncio.apply()
+            check_patch_tornado()
```

Comment:
We might want to do that only once if we had a loop, not at each coroutine execution?

Reply:
Well, the patch is not applied when tornado is not imported (this happens in vaex, where I first call this function, then start the server, and call this function again). Also the patch is only applied once (it checks if the Future is in the set).

Reply:
OK, but what about `check_ipython()` and `nest_asyncio.apply()`?

Reply:
For check_ipython the same holds (it can be imported later on), and nest_asyncio has a check build in, in the end, simplicity, otherwise we need to check things that nest_asyncio already checks.